##################################################
#Parameters to configure the neccesary directories
##################################################

directories_config:
  open_pose_python: "/home/leo/catkin_ws/src/openpose/build/python"
  open_pose_models: "/home/leo/catkin_ws/src/openpose/models"
  gesture_classifier_model: "/home/leo/catkin_ws/src/robocup_human_sensing/config/classifier_model_3D_v6.joblib"
  
##############################################################################################################################################################
#Parameters to calibrate human detection, human tracking algorithms
##############################################################################################################################################################

camera_config:
  intr_param: [384.7431945800781, 326.4798278808594, 384.34613037109375, 244.670166015625] # intrinsic parameters of realsense D455 such that -> [fx cx fy cy] 
  camera_topic_names: ["/camera/color/image_raw","/camera/aligned_depth_to_color/image_raw"] # name of the topics published by the Tiago camera [color,depth] 
  
openpose_config:
  openpose_normal_performance: "-1x160" # net resolution of the openpose when working on "normal" performance mode, has to be numbers multiple of 16, the default is "-1x368", and the fastest performance is "-1x160", used for human presence monitoring, "-1x320"
  openpose_high_performance: "-1x240" # net resolution of the openpose when working on "high" performance mode, used for gesture recognition, "-1x480"
  skeleton_extract_param: 19 #number of skeleton joints (openpose output) considered important for gesture detection algorithm, 19 means that feet joints are not considered
  joints_min: 4 # minimum number of joints in the skeleton detected that make it a valid detection
  confidence_min: 0.6 # minimim confidence factor used to consider a skeleton detected by Openpose valid or not
  
